### PR TITLE
Wazo 816 trunk name username

### PR DESCRIPTION
**Pull request by pc-m**

### CODE REVIEW

**Comment by pc-m** [comment]
Depends-On: https://github.com/wazo-platform/xivo-dao/pull/12

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,14 @@
 Changelog
 =========
 
+19.15
+-----
+
+* The POST `/1.1/endpoints/sip` now accepts a `name` and a `username` field. If
+  you relied on the behavior that copied to username to the name field you must
+  upgrade your before that compatibility layer gets removed.
```
Comment (fblackburn1) [inline CHANGELOG.md]: `upgrade your` what?

File: wazo_confd/plugins/endpoint_sip/schema.py [hunk 2]
```diff
@@ -26,6 +27,20 @@ class SipSchema(BaseSchema):
     trunk = fields.Nested('TrunkSchema', only=['id', 'links'], dump_only=True)
     line = fields.Nested('LineSchema', only=['id', 'links'], dump_only=True)
 
+    @post_load
+    def set_name_to_username_if_missing(self, data, **kwargs):
+        name = data.get('name')
+        if not name and 'username' in data:
+            data['name'] = data.get('username')
```
Comment (fblackburn1) [inline wazo_confd/plugins/endpoint_sip/schema.py]:
```suggestion
            data['name'] = data['username']
```

**Comment by fblackburn1** [comment]
Also can you add a comment in the code (in the post_load/post_dump) or rename method to explicitly mark as a deprecated compatibility layer. Then when we will remove it, we can isolate the code easily

**Comment by fblackburn1** [comment]
Bamm !!! In your face zuul :middle_finger:

**Pull request merged**